pkg/snapstore/swift_snapstore_test.go

```diff
@@ -150,3 +178,50 @@ func handleDeleteObject(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 	}
 }
+
+// handleBulkDeleteObject creates an HTTP handler at `/testContainer/testObject` on the test handler mux that
+// responds with a `Delete` response.
+func handleBulkDeleteObject(w http.ResponseWriter, r *http.Request) {
+	objectMapMutex.Lock()
+	defer objectMapMutex.Unlock()
+
+	var jsonResponse bulkDeleteJSONResponse
+
+	buf := new(bytes.Buffer)
+	if _, err := io.Copy(buf, r.Body); err != nil {
+		errorMessage := fmt.Sprintf("failed to read content %v", err)
+		logrus.Errorf(errorMessage)
+		jsonResponse.ErrorStrings = append(jsonResponse.ErrorStrings, errorMessage)
+		w.WriteHeader(http.StatusOK)
+		return
+	}
+
+	segmentObjects := strings.Split(string(buf.Bytes()), "\n")
+	// last string is an empty string, artefact of '\n' at the end
+	segmentObjects = segmentObjects[:len(segmentObjects)-1]
+	for _, segmentObject := range segmentObjects {
+		segmentObject = "/" + segmentObject
+		// objects.BulkDelete() internally calls url.QueryEscape
+		unescapedQueryForObject, err := url.QueryUnescape(segmentObject)
+		if err != nil {
+			errorMessage := fmt.Sprintf("failed to unescape url %v", err)
+			logrus.Errorf(errorMessage)
+			jsonResponse.ErrorStrings = append(jsonResponse.ErrorStrings, errorMessage)
+		}
+
+		key := parseObjectNamefromURL(&url.URL{Path: unescapedQueryForObject})
+		if _, ok := objectMap[key]; ok {
+			delete(objectMap, key)
+			jsonResponse.Deleted++
+		} else {
+			errorMessage := fmt.Sprintf("Resource not found: %s", segmentObject)
+			logrus.Errorf(errorMessage)
+			jsonResponse.NotFound++
+			jsonResponse.ErrorStrings = append(jsonResponse.ErrorStrings, errorMessage)
+		}
+	}
+
+	marshalledResponse, _ := json.Marshal(jsonResponse)
+	w.WriteHeader(http.StatusOK)
```

Comment:
Use an array of errors called `errs` and append each collected error to that. At the end, if there are any errors present, then write header with an error response code, based on what kind of error (notFound, internal error, etc). Additionally, set response `Err` to errors.Join(errs), which allows you to join all the collected errors into a single error object.

The BulkDelete we use does not consume the response Body, so it may not make much sense to write anything to it, but rather just log the results like `total segments`, `found`, `not found`, `deleted`, etc

Reply:
Will just log the errors that occur, since the body of the response is not consumed anyway.

Should the response code be something other than 200, since Swift always returns 200 as explained in https://github.com/gardener/etcd-backup-restore/pull/710#discussion_r1512209890